torch_em/model/probabilistic_unet.py

```diff
@@ -0,0 +1,445 @@
+# This code is based on the original TensorFlow implementation: https://github.com/SimonKohl/probabilistic_unet
+# The below implementation is from: https://github.com/stefanknegt/Probabilistic-Unet-Pytorch
+
+import numpy as np
+
+import torch
+import torch.nn as nn
+from torch.distributions import Normal, Independent, kl
+
+from torch_em.model import UNet2d
+from torch_em.loss.dice import DiceLossWithLogits
+
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+
+
+def truncated_normal_(tensor, mean=0, std=1):
+    size = tensor.shape
+    tmp = tensor.new_empty(size + (4,)).normal_()
+    valid = (tmp < 2) & (tmp > -2)
+    ind = valid.max(-1, keepdim=True)[1]
+    tensor.data.copy_(tmp.gather(-1, ind).squeeze(-1))
+    tensor.data.mul_(std).add_(mean)
+
+
+def init_weights(m):
+    if type(m) == nn.Conv2d or type(m) == nn.ConvTranspose2d:
+        nn.init.kaiming_normal_(m.weight, mode='fan_in', nonlinearity='relu')
+        # nn.init.normal_(m.weight, std=0.001)
+        # nn.init.normal_(m.bias, std=0.001)
+        truncated_normal_(m.bias, mean=0, std=0.001)
+
+
+def init_weights_orthogonal_normal(m):
+    if type(m) == nn.Conv2d or type(m) == nn.ConvTranspose2d:
+        nn.init.orthogonal_(m.weight)
+        truncated_normal_(m.bias, mean=0, std=0.001)
+        # nn.init.normal_(m.bias, std=0.001)
+
+
+class Encoder(nn.Module):
+    """
+    A convolutional neural network, consisting of len(num_filters) times a block of no_convs_per_block
+    convolutional layers, after each block a pooling operation is performed.
+    And after each convolutional layer a non-linear (ReLU) activation function is applied.
+    """
+
+    def __init__(
+                self,
```

Comment:
Just a minor thing, but the usual style for these inits is to have only one indent:
```
def __init__(
    self,
    aa,
    bb,
    cc,
    dd,
    ...
):
   ...
```
(also relevant in several places below).